pkg/resourcecollector/resourcetransformation.go

```diff
@@ -0,0 +1,138 @@
+package resourcecollector
+
+import (
+	"fmt"
+	"strings"
+
+	stork_api "github.com/libopenstorage/stork/pkg/apis/stork/v1alpha1"
+	storkops "github.com/portworx/sched-ops/k8s/stork"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// Since we collect all resources from required migration namespace at once
+// getResourcePatch creates map of namespace: {kind: []resourceinfo{}}
+// to get transform spec for matching resources
+func GetResourcePatch(transformName string, namespaces []string) (map[string]stork_api.KindResourceTransform, error) {
+	// namespace- Kind:TransformSpec map for faster lookup
+	patch := make(map[string]stork_api.KindResourceTransform)
+	if transformName == "" {
+		logrus.Infof("empty name ")
+		return patch, nil
+	}
+	for _, namespace := range namespaces {
+		resp, err := storkops.Instance().GetResourceTransformation(transformName, namespace)
+		if err != nil && !errors.IsNotFound(err) {
+			logrus.Infof("not found in namespace: %v", transformName)
+			return nil, err
+		}
+		resMap := make(map[string][]stork_api.TransformResourceInfo)
+		for _, resource := range resp.Status.Resources {
+			resMap[resource.Kind] = append(resMap[resource.Group], *resource)
+		}
+		patch[namespace] = resMap
+	}
+	logrus.Infof("resource patch : %v", patch)
+	return patch, nil
+}
+
+// this method transform object as per resource transformation specified in each namespaces
+func TransformResources(
+	object runtime.Unstructured,
+	resPatch []stork_api.TransformResourceInfo,
+	name, namespace string,
```

Comment:
nit: lets rename these to objName and objNamespace. Also we should be able to determine the name and namespace from the unstructured object right?